### PR TITLE
Added new command-line option `--compiler-opts`.

### LINKED ISSUE #189: pass in compiler-opts on commandline
Depending on host OS and installed toolchains we need to pass in different compiler options to make various ffigen invocations work.

Instead of adding all individual versions to the `yaml` configuration, it might make sense to allow passing these options in to the ffigen invocation.

Instead of

```yaml
compiler-opts: '-I/usr/local/opt/llvm/lib/clang/11.1.0/include/' # 11.1.0 might be some other version
```

write a script

```dart
import 'dart:io';

void main(List<String> arguments) {
  final headers = headerPaths();
  final root =
      Directory.fromUri(Platform.script).parent.parent.uri.toFilePath();
  final result = Process.runSync('dart', [
    'run',
    'ffigen',
    '--config=$root/bin/ffigen.yaml',
    ...headers, // This is currently not supporteed
  ]);
}

List<String> headerPaths() {
  if (Platform.isMacOS) {
    const brewLlvmPath = "/usr/local/opt/llvm/lib/clang";
    final result = Process.runSync('ls', [brewLlvmPath]);
    final stdout = result.stdout as String;
    if (stdout == "") {
      print("The stdlib headers could not be found.");
      print("Please run `brew install llvm`.");
      exit(1);
    }

    // We pick the first available one.
    final version = stdout.split("\n").first;

    final path = "$brewLlvmPath/$version/include";
    final include = "-I$path";
    return [include];
  }
  throw UnsupportedError("Unknown operating system");
}
```

(Without providing headers `stdarg.h` and other headers from the standard library cannot be found by clang on various project.)

Maybe we should interpret all arguments that we don't consume as `compiler-opts`.

### PR DESCRIPTION
**Pull request by mannprerak2**
Closes #189 
- Added new command-line option --compiler-opts.
- Added test, update changelog, version, readme.

### CODE REVIEW

**Comment by mannprerak2** [comment]
@dcharkes should I also resolve #190 here?

**Review by dcharkes — APPROVED** [review]
lgtm

**Comment by dcharkes** [comment]
> @dcharkes should I also resolve #190 here?

Whatever you prefer, a separate PR is fine for me as well.

**Comment by mannprerak2** [comment]
Let's do a separate PR then, I hadn't done any work on that till now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# 2.2.5
+- Added new command line flag `--compiler-opts` to ffigen.
```
Comment (dcharkes) [inline CHANGELOG.md:2]: to ffigen. -> to the command line tool.

Comment (mannprerak2) [reply CHANGELOG.md]: Done

**Pull request merged**